crates/shell/src/shells/fish.rs

```diff
@@ -57,8 +57,91 @@ end;"#,
         .into_iter()
         .collect()
     }
-}
 
+    /// Quotes a string according to Fish shell quoting rules.
+    ///
+    /// This method handles quoting and escaping according to Fish shell rules.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - The string to be quoted.
+    ///
+    /// # Returns
+    ///
+    /// A quoted string suitable for use in Fish shell scripts.
+    fn quote(&self, value: &str) -> String {
+        if value.is_empty() {
+            return "''".to_string();
+        }
+
+        // Check for complex values requiring special quoting
+        if value.contains('\n')
```

Comment:
Gotta be a easier way to do this... hrmmm. Maybe an array? Which can then be used in the replace also?